src/main/java/org/mitre/synthea/modules/BloodPressureValueGenerator.java

```diff
@@ -47,156 +87,167 @@ public BloodPressureValueGenerator(Person person, SysDias sysDias) {
+        if (hypertension) {
+          baseline = person.rand(HYPERTENSIVE_DIA_BP_RANGE);
+        } else {
+          baseline = person.rand(NORMAL_DIA_BP_RANGE);
+        }
+      }
+
+      if ("M".equals(person.attributes.get(Person.GENDER))) {
+        baseline += 1;
+      } else {
+        baseline -= 1;
       }
+
+      person.attributes.put(bpBaselineKey, baseline);
+    }
+    return baseline;
+  }
+
+  private static final long ONE_YEAR = Utilities.convertTime("years", 1);
+
+  private double getLifestyleImpacts(Person person, double baseline, long time) {
+    // if the person has a "blood pressure care plan"
+    // assume that over ~1 year their blood pressure will be reduced by ~14 mmHg
+    // with most of that happening in the first 6 mos
+
+    // https://pubmed.ncbi.nlm.nih.gov/12709466/
+
+    double maxDrop;
+    if (this.sysDias == SysDias.SYSTOLIC) {
+      // don't allow diet/exercise alone to drop below 124
+      double delta = Math.abs(baseline - 124);
+      maxDrop = -Math.min(delta, 14.0);
     } else {
-      if (hypertension) {
-        if (!bloodPressureControlled) {
-          return person.rand(HYPERTENSIVE_DIA_BP_RANGE);
+      // don't allow diet/exercise alone to drop below 78
+      double delta = Math.abs(baseline - 78);
+      maxDrop = -Math.min(delta, 8.0);
+    }
+
+    // 443402002 = base hypertension careplan
+    // allow for multiple codes, and consider each independently,
+    // but if one is active then skip the rest
+    // (assume they don't add together)
+    String[] carePlanCodes = {"443402002"};
+
+    for (String code : carePlanCodes) {
+      double adherenceRatio = 0.15;
+      // estimate 3-25% of patients are adherent,
+      // so for a single # pick ~15%
+
+      HealthRecord.CarePlan careplan = (HealthRecord.CarePlan) person.record.present.get(code);
+
+      if (careplan != null && careplan.stop == 0L) {
+        boolean carePlanAdherent;
+        String key = "htn_trial_" + code + "_careplan_adherent";
+        if (person.attributes.containsKey(key)) {
+          carePlanAdherent = (boolean) person.attributes.get(key);
         } else {
-          return person.rand(NORMAL_DIA_BP_RANGE);
+          carePlanAdherent = person.rand() < adherenceRatio;
+          person.attributes.put(key, carePlanAdherent);
+        }
+        if (carePlanAdherent) {
+          long start = careplan.start;
+
+          if (time > (start + ONE_YEAR)) {
+            // max value
+            return maxDrop;
+          } else if (time < start) {
+            return 0.0;
+          }
+
+          // dy / dx = -14/1 = -14
+          // y = (dy/dx) * x
+          // x = (time - start) / 1 yr
+
+          double x = ((double)(time - start) / ((double)ONE_YEAR));
+
+          return x * maxDrop;
         }
-      } else {
-        return person.rand(NORMAL_DIA_BP_RANGE);
       }
     }
+
+    return 0.0;
+  }
+
+  private double getMedicationImpacts(Person person, long time) {
```

Comment:
The argument of `time` will not be used once it is removed from `getDrugImpact` where it is also not used.